mp/src/materialsystem/stdshaders/pbr_dx9.cpp

```diff
@@ -0,0 +1,620 @@
+    END_SHADER_PARAMS;
+
+    // Setting up variables for this shader
+    void SetupVars(PBR_Vars_t &info)
+    {
+        info.baseTexture = BASETEXTURE;
+        info.baseColor = COLOR;
+        info.normalTexture = NORMALTEXTURE;
+        info.bumpMap = BUMPMAP;
+        info.baseTextureFrame = FRAME;
+        info.baseTextureTransform = BASETEXTURETRANSFORM;
+        info.alphaTestReference = ALPHATESTREFERENCE;
+        info.flashlightTexture = FLASHLIGHTTEXTURE;
+        info.flashlightTextureFrame = FLASHLIGHTTEXTUREFRAME;
+        info.envMap = ENVMAP;
+        info.emissionTexture = EMISSIONTEXTURE;
+        info.mraoTexture = MRAOTEXTURE;
+        info.useEnvAmbient = USEENVAMBIENT;
+        info.specularTexture = SPECULARTEXTURE;
+        info.useParallax = PARALLAX;
+        info.parallaxDepth = PARALLAXDEPTH;
+        info.parallaxCenter = PARALLAXCENTER;
+    };
+
+    // Initializing parameters
+    SHADER_INIT_PARAMS()
+    {
+        // Fallback for changed parameter
+        if (params[NORMALTEXTURE]->IsDefined())
+            params[BUMPMAP]->SetStringValue(params[NORMALTEXTURE]->GetStringValue());
+
+        // Dynamic lights need a bumpmap
+        if (!params[BUMPMAP]->IsDefined())
+            params[BUMPMAP]->SetStringValue("dev/flat_normal");
+
+        // PBR relies heavily on envmaps
+        if (!params[ENVMAP]->IsDefined())
+            params[ENVMAP]->SetStringValue("env_cubemap");
+
+        // Check if the hardware supports flashlight border color
+        if (g_pHardwareConfig->SupportsBorderColor())
+        {
+            params[FLASHLIGHTTEXTURE]->SetStringValue("effects/flashlight_border");
+        }
+        else
+        {
+            params[FLASHLIGHTTEXTURE]->SetStringValue("effects/flashlight001");
+        }
+    };
+
+    // Define shader fallback
+    SHADER_FALLBACK
+    {
+        return 0;
+    };
+
+    SHADER_INIT
+    {
+        PBR_Vars_t info;
+        SetupVars(info);
+
+        Assert(info.flashlightTexture >= 0);
+        LoadTexture(info.flashlightTexture, TEXTUREFLAGS_SRGB);
+
+        Assert(info.bumpMap >= 0);
+        LoadBumpMap(info.bumpMap);
+
+        Assert(info.envMap >= 0);
+        int envMapFlags = g_pHardwareConfig->GetHDRType() == HDR_TYPE_NONE ? TEXTUREFLAGS_SRGB : 0;
+        envMapFlags |= TEXTUREFLAGS_ALL_MIPS;
+        LoadCubeMap(info.envMap, envMapFlags);
+
+        if (info.emissionTexture >= 0 && params[EMISSIONTEXTURE]->IsDefined())
+            LoadTexture(info.emissionTexture, TEXTUREFLAGS_SRGB);
+
+        Assert(info.mraoTexture >= 0);
+        LoadTexture(info.mraoTexture, 0);
+
+        //bool bIsBaseTextureTranslucent = false;
+        if (params[info.baseTexture]->IsDefined())
+        {
+            LoadTexture(info.baseTexture, TEXTUREFLAGS_SRGB);
+
+            //if (params[info.baseTexture]->GetTextureValue()->IsTranslucent())
+            //{
+            //    bIsBaseTextureTranslucent = true;
+            //}
+        }
```

Comment:
Should this commented code still be here?

Reply:
I think this used to be for $translucent or something but isn't necessary anymore, I'll remove it

Reply:
Alright